src/samplers/multinomial.jl

```diff
@@ -1,33 +1,33 @@
-function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{Float64},
-                         x::AbstractVector{<:Real})
+function multinom_rand!(rng::AbstractRNG, n::Int, p::AbstractVector{Tp},
+                         x::AbstractVector{Tx}) where {Tx <: Real, Tp <: Real}
     k = length(p)
     length(x) == k || throw(DimensionMismatch("Invalid argument dimension."))
 
-    rp = 1.0  # remaining total probability
+    rp = Tp(1.0)  # remaining total probability
     i = 0
     km1 = k - 1
 
     while i < km1 && n > 0
         i += 1
         @inbounds pi = p[i]
         if pi < rp
-            xi = rand(rng, Binomial(n, pi / rp))
+            xi = Tx(rand(rng, Binomial(n, Float64(pi / rp))))
             @inbounds x[i] = xi
-            n -= xi
+            n -= Int64(xi)
```

Comment:
Not needed, as `rand` above returns an `Int` (generally, also one should not use `Int64` explicitly if there is not a very good reason - on 32bit machines you end up with a (very slow) mix of `Int32` and `Int64` otherwise):
```suggestion
            n -= xi
```